src/main/java/com/project/food_ordering_service/domain/delivery/service/DeliveryService.java

```diff
@@ -0,0 +1,95 @@
+package com.project.food_ordering_service.domain.delivery.service;
+
+import com.project.food_ordering_service.domain.delivery.entity.Delivery;
+import com.project.food_ordering_service.domain.delivery.exception.DeliveryNotFoundException;
+import com.project.food_ordering_service.domain.delivery.repository.DeliveryRepository;
+import com.project.food_ordering_service.domain.order.entity.Order;
+import com.project.food_ordering_service.domain.order.entity.OrderStatus;
+import com.project.food_ordering_service.domain.order.exception.OrderNotFoundException;
+import com.project.food_ordering_service.domain.order.repository.OrderRepository;
+import com.project.food_ordering_service.domain.user.entity.Role;
+import com.project.food_ordering_service.domain.user.entity.User;
+import com.project.food_ordering_service.domain.user.exception.UserNotFoundException;
+import com.project.food_ordering_service.domain.user.repository.UserRepository;
+import lombok.RequiredArgsConstructor;
+import org.springframework.data.domain.Page;
+import org.springframework.data.domain.Pageable;
+import org.springframework.stereotype.Service;
+import org.springframework.transaction.annotation.Transactional;
+
+@Service
+@RequiredArgsConstructor
+public class DeliveryService {
+
+    private final DeliveryRepository deliveryRepository;
+    private final OrderRepository orderRepository;
+    private final UserRepository userRepository;
+
+    @Transactional
+    public Delivery assignDelivery(Long orderId, Long riderId) {
+        Order order = orderRepository.findById(orderId)
+                .orElseThrow(OrderNotFoundException::new);
+
+        if (order.getStatus() != OrderStatus.DELIVERY_REQUESTED) {
+            throw new IllegalStateException("배달 요청 상태가 올바르지 않습니다.");
+        }
+
+        User rider = userRepository.findById(riderId)
+                .orElseThrow(UserNotFoundException::new);
+
+        if (rider.getRole() != Role.RIDER) {
+            throw new IllegalArgumentException("배달원만 배달을 할 수 있습니다.");
+        }
+
+        Delivery delivery = Delivery.builder()
+                .order(order)
+                .rider(rider)
+                .status(OrderStatus.COMPLETED) // 배달원이 배달을 수락한 상태
+                .build();
+
+        order.updateOrderStatus(OrderStatus.ONTHEWAY);
```

Comment:
```suggestion
//여기 부분
  order.updateOrderStatus(OrderStatus.ONTHEWAY);
  
 Delivery delivery = Delivery.builder()
                .order(order)
                .rider(rider)
                .status(OrderStatus.COMPLETED) // 배달원이 배달을 수락한 상태
                .build();

```
위에 "여기부분" 주문 상태를 변경값을 클라이언트에 받아왔는데 정적으로 상태 변경 하신 이유가 있나요?
그리고 주문 상태를 먼저 변경 하고 새로운 인스턴스를 생성 해야 변경된 값이 들어가는거 아닌가요?
확인해보셨으면 어떤 주문 상태가 Delivery 저장 되는건지 이야기 해주세요!
@mistyblue0302

Reply:
@RudySeo 
Delivery 객체와 Order 객체의 상태 일관성을 위해 수정하였습니다! 주문 상태 변경 후 그 상태를 기반으로 배달 정보를 생성하는 것이 맞는 것 같습니다

assingDelivery() 메소드는 배달원이 배달요청 상태인지 확인 후 수락하는 역할을 합니다! 배달 상태 업데이트는 updateDeliveryStatus() 메소드 안에서 이루어집니다.

주문이 배달원에 의해 수락되었음을 나타내기 위해 정적으로 상태를 변경하였습니다!!